app/scripts/snaps/preinstalled-snaps.ts

```diff
@@ -0,0 +1,53 @@
+import type { PreinstalledSnap } from '@metamask/snaps-controllers';
+
+// Preinstalled snaps requires fs and require to read and use any installed snaps.
+// We can switch `require` to `import`, but then this file gets transpiled to ESM & won't have access to `require.resolve`.
+// Work is needed to create the require object in ESM - specifically the import url. See snippet
+// import { createRequire } from 'node:module'
+// const require = createRequire(import.meta.url)
+//                                ^ The 'import.meta' meta-property is not allowed in files which will build into CommonJS output
+/* eslint-disable-next-line @typescript-eslint/no-require-imports,@typescript-eslint/no-var-requires */
+const fs = require('fs');
+
+const TEMP_PREINSTALLED_SNAPS = [];
```

Comment:
Do we still need this temporary variable now that we removed the try-catch?

Reply:
The temp is used since the push/inserting of snaps is done within the codefence.
Can we convert a normal array into a frozen object?

Reply:
Update tested, that is interesting - TIL Object.freeze mutates original!

[Playground Link](https://www.typescriptlang.org/play?#code/MYewdgzgLgBADgJwKYEtJQIYBstICYDKYGcEAXDNAmgOYDaAujALwyMDcAUJ4qutrkLFSAOjgBXCAAsAFAHIIIALZJKwuQEpuodDGRRxCMPgBq2FjADyAIwBWSYFBEAzZEgBeSGbzTQB+IhIILU4AelCYMBAAdxho1WAMMDlYCWkYKBB4ZF9MHHw1IJgkvAypVX1DY1KAN2xxJB4c-nyhILFJWTk8ECQIMFhokAQAa01OSqNTbA7peWMUKHKEGB6+spQIABoYa3FYAElVrJokWAwMgE84VSQEBGGYZdUyABFNIA)